kpatch-build/create-diff-object.c

```diff
@@ -3715,6 +3830,42 @@ static void kpatch_create_mcount_sections(struct kpatch_elf *kelf)
 		}
 
 		switch(kelf->arch) {
+		case AARCH64: {
+			struct section *sec;
+			unsigned char *insn;
+			int i;
+
+			insn = sym->sec->data->d_buf;
+
+			/*
+			 * If BTI (Branch Target Identification) is enabled then there
+			 * might be an additional 'BTI C' instruction before the two
+			 * patchable function entry 'NOP's.
+			 * i.e. 0xd503245f (little endian)
+			 */
+			if (insn[0] == 0x5f) {
+				if (insn[1] != 0x24 || insn[2] != 0x03 || insn[3] != 0xd5)
+					ERROR("%s: unexpected instruction in patch section of function\n", sym->name);
+				insn_offset += 4;
+				insn += 4;
+			}
+			for (i = 0; i < 8; i += 4) {
+				/* We expect a NOP i.e. 0xd503201f (little endian) */
+				if (insn[i] != 0x1f || insn[i + 1] != 0x20 ||
+				    insn[i + 2] != 0x03 || insn [i + 3] != 0xd5)
+					ERROR("%s: unexpected instruction in patch section of function\n", sym->name);
+			}
+
+			/* Allocate __patchable_function_entries for symbol */
+			sec = create_section_pair(kelf, "__patchable_function_entries", sizeof(void *), 1);
+			sec->sh.sh_flags |= SHF_WRITE | SHF_LINK_ORDER;
+			/* We will reset this sh_link in the reindex function. */
+			sec->sh.sh_link = 0;
+
```

Comment:
A FYI for this change: 
Thought I am emitting multiple "pfe" sections for every single functions here. In the later step in `kpatch-build`, the `$(LD) -r output.o <input.o files>` might still merge these pfe sections into one. At least this is current (v2.41) behavior for `ld` and `ld.gold` . 
For `ld.lld`, it will keep these multiple pfe sections as separated. 

But I think it is not a big issue here. The pfe sections will be squashed into one anyway in the finally ko file when building the `/kmod/patch`.